app/modules/provisioning/groups.py

```diff
@@ -40,3 +46,11 @@ def get_groups_with_members_from_integration(integration_source, **kwargs):
     for filter in filters:
         groups = filter_tools.filter_by_condition(groups, filter)
     return groups
+
+
+def preformat_groups(groups, lookup_key, new_key, find="", replace=""):
+    for group in groups:
+        if lookup_key not in group:
+            raise KeyError(f"Group {group} does not have {lookup_key} key")
+        group[new_key] = group[lookup_key].replace(find, replace)
```

Comment:
Could we tighten this so that it only replaces the first instance of `find` token if it is a prefix?  


```sh
# This would get replaced
AWS-some-group-name

# But not this
some-group-AWS-name
```

Reply:
Newest commit fixes this issue!